lib/buildenvsetup/ceconan.ts

```diff
@@ -76,7 +76,7 @@ export class BuildEnvSetupCeConanDirect extends BuildEnvSetupBase {
             };
 
             request(url, settings, (err, res, body) => {
-                if (res.statusCode === 404) {
+                if (res && res.statusCode === 404) {
```

Comment:
I tihnk we should check `err` first: I suspect that's what is going on. My mistake! I don't think it's valid to be called with `err` _and_ `res` null, and typically this is the pattern I've seen,

Reply:
Is `@types/request` wrong? It shouldn't be possible for `res` to be null or undefined

Reply:
oh hmm - if there's an error what would `request` pass as the `res` ?

Reply:
though you're right, types says:
```
    type RequestCallback = (error: any, response: Response, body: any) => void;
```

Reply:
but the original library is not typescript, the type hints are an approximation in the ideal case

Reply:
Agreed: I think @jeremy-rifkin 's point is "maybe the library is wrong" and I think this is evidence it is :)

Reply:
all 3rd party libraries have issues, just different ones we haven't found yet

Reply:
Yeah, incorrect types aren't too shocking. When we turn `@typescript-eslint/no-unnecessary-condition` on for lib/ checking if `res` is truthy will be a problem. I might make a PR for @types/request but I'm not sure they'll merge since it's deprecated.

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63804 filed @jeremy-rifkin (the filing process seemed to create a discussion instead of an issue)